internal/restorer/filerestorer.go

```diff
@@ -277,37 +269,36 @@ func (r *fileRestorer) downloadPack(ctx context.Context, pack *packInfo) {
 						// - should allow concurrent writes to the file
 						// so write the first blob while holding file lock
 						// write other blobs after releasing the lock
-						file.lock.Lock()
-						create := file.flags&fileProgress == 0
 						createSize := int64(-1)
-						if create {
+						file.lock.Lock()
+						if file.inProgress {
+							file.lock.Unlock()
+						} else {
 							defer file.lock.Unlock()
-							file.flags |= fileProgress
+							file.inProgress = true
 							createSize = file.size
-						} else {
-							file.lock.Unlock()
 						}
 						return r.filesWriter.writeToFile(r.targetPath(file.location), blobData, offset, createSize)
 					}
-					err := writeToFile()
+					err := sanitizeError(file, writeToFile())
```

Comment:
That change will now ignore file write errors. But I guess there's no harm in trying to restore all other occurrences of a blob?

Reply:
Yes, I think trying is worth it :wink: